Library/Homebrew/formula.rb

```diff
@@ -872,39 +872,14 @@ def with_logging(log_type)
     @active_log_type = old_log_type
   end
 
-  # This method can be overridden to provide a plist.
-  # @see https://www.unix.com/man-page/all/5/plist/ Apple's plist(5) man page
-  # <pre>def plist; <<~EOS
-  #  <?xml version="1.0" encoding="UTF-8"?>
-  #  <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-  #  <plist version="1.0">
-  #  <dict>
-  #    <key>Label</key>
-  #      <string>#{plist_name}</string>
-  #    <key>ProgramArguments</key>
-  #    <array>
-  #      <string>#{opt_bin}/example</string>
-  #      <string>--do-this</string>
-  #    </array>
-  #    <key>RunAtLoad</key>
-  #    <true/>
-  #    <key>KeepAlive</key>
-  #    <true/>
-  #    <key>StandardErrorPath</key>
-  #    <string>/dev/null</string>
-  #    <key>StandardOutPath</key>
-  #    <string>/dev/null</string>
-  #  </dict>
-  #  </plist>
-  #  EOS
-  # end</pre>
-  def plist
-    nil
-  end
-
   # The generated launchd {.plist} service name.
   def plist_name
-    "homebrew.mxcl." + name
+    self.class.plist_name("homebrew.mxcl." + name)
+  end
+
+  # @private
+  def self.plist_name(label)
```

Comment:
How is this function useful?

Reply:
@reitermarkus Because without it I get an `undefined method `plist_name'` error… and doing it this way (essentially inverting the behaviour of the previous implementation) I have access to `name` which is now passed from the instance method to the class method and everything works, without having to resort to messy hacks like we had before.

Reply:
Please move all `self.` methods to under `class << self` in `formula.rb` to be consistent.